### PR TITLE
[SPARK-40973][SQL] Rename `_LEGACY_ERROR_TEMP_0055` to `UNCLOSED_BRACKETED_COMMENT`

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to introduce new error class `UNCLOSED_BRACKETED_COMMENT`, by updating the existing legacy temp error class `_LEGACY_ERROR_TEMP_0055 `.

### Why are the changes needed?

We should use appropriate error class name that matches the error message.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

The existing CI should pass.

### CODE REVIEW

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1418,7 +1418,7 @@
       "It is not allowed to add database prefix `<database>` for the TEMPORARY view name."
     ]
   },
-  "_LEGACY_ERROR_TEMP_0055" : {
+  "UNCLOSED_BRACKETED_COMMENT" : {
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:1421]: Don't think you can rename in place. Error class names should be sorted in the file.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (itholic) [reply core/src/main/resources/error/error-classes.json]: Oops, my mistake. Thanks for the correction!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala [hunk 2]
```diff
@@ -608,8 +608,12 @@ private[sql] object QueryParsingErrors extends QueryErrorsBase {
   }
 
   def unclosedBracketedCommentError(command: String, position: Origin): Throwable = {
-    new ParseException(Some(command), "Unclosed bracketed comment", position, position,
-      Some("_LEGACY_ERROR_TEMP_0055"))
+    new ParseException(
+      Some(command),
+      "Found an unclosed bracketed comment. Please, append */ at the end of the comment.",
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:613]: Please, don't duplicate the error message here.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala (hunk 2 above)
Comment (itholic) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:613]: I think maybe If we don't specify the message here, then `null` is returned so the `CliSuite.SPARK-37555: spark-sql should pass last unclosed comment to backend` failed as below:

```
CliSuite.SPARK-37555: spark-sql should pass last unclosed comment to backend
org.scalatest.exceptions.TestFailedException: 
=======================
CliSuite failure output
=======================
Spark SQL CLI command line: ../../bin/spark-sql --master local --driver-java-options -Dderby.system.durability=test --conf spark.ui.enabled=false --hiveconf javax.jdo.option.ConnectionURL=jdbc:derby:;databaseName=/home/runner/work/spark/spark/target/tmp/spark-346cf558-8a95-43f1-b893-81f6a5571f3f;create=true --hiveconf hive.exec.scratchdir=/home/runner/work/spark/spark/target/tmp/spark-f108bccb-3f71-4854-aa70-c18ea721617c --hiveconf conf1=conftest --hiveconf conf2=1 --hiveconf hive.metastore.warehouse.dir=/home/runner/work/spark/spark/target/tmp/spark-86d39e14-55f9-437b-a676-7a5cabd58904
Exception: java.util.concurrent.TimeoutException: Futures timed out after [2 minutes]
Failed to capture next expected output "Found an unclosed bracketed comment." within 2 minutes.

2022-10-31 19:36:30.434 - stderr> Setting default log level to "WARN".
2022-10-31 19:36:30.434 - stderr> To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
2022-10-31 19:36:41.979 - stderr> Spark master: local, Application Id: local-1667270193279
2022-10-31 19:36:43.351 - stdout> spark-sql> /* SELECT /*+ HINT() 4; */;
2022-10-31 19:36:44.119 - stderr> 
2022-10-31 19:36:44.119 - stderr> [PARSE_SYNTAX_ERROR] Syntax error at or near ';'(line 1, pos 26)
2022-10-31 19:36:44.119 - stderr> 
2022-10-31 19:36:44.119 - stderr> == SQL ==
2022-10-31 19:36:44.119 - stderr> /* SELECT /*+ HINT() 4; */;
2022-10-31 19:36:44.119 - stderr> --------------------------^^^
2022-10-31 19:36:44.119 - stderr> 
2022-10-31 19:36:44.141 - stdout> spark-sql> /* SELECT /*+ HINT() 4; */ SELECT 1;
2022-10-31 19:36:48.341 - stdout> 1
2022-10-31 19:36:48.341 - stderr> Time taken: 4.196 seconds, Fetched 1 row(s)
2022-10-31 19:36:48.352 - stdout> spark-sql> /* Here is a unclosed bracketed comment SELECT 1;
2022-10-31 19:36:48.361 - stderr> 
2022-10-31 19:36:48.361 - stderr> null(line 1, pos 0)
```

I think maybe we might need to introduce new `SparkException` for `ParseException` such as `SparkParseException` ??

So, can we keep it as is for now, and remove the duplicated message when `SparkParseException` is introduced ??

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala [hunk 3]
```diff
@@ -608,8 +608,12 @@ private[sql] object QueryParsingErrors extends QueryErrorsBase {
...
+      position,
+      position,
+      Some("UNCLOSED_BRACKETED_COMMENT"))
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:616]: Could you replace this by:
```suggestion
    new ParseException(
      command = Some(command),
      start = start,
      stop = stop,
      errorClass = "UNCLOSED_BRACKETED_COMMENT",
      messageParameters = Map.empty)
```
and invoke via:
```scala
      throw QueryParsingErrors.unclosedBracketedCommentError(
        command = command,
        start = Origin(Option(failedToken.getStartIndex)),
        stop = Origin(Option(failedToken.getStopIndex)))
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala (hunk 3 above)
Comment (itholic) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:616]: Cool! Thanks for the suggestion 👍

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Comment by itholic** [comment]
Oh, `SparkThrowableSuite` failed. Let me take a look, soon.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @itholic.

**Pull request closed without merging**